src/main/host/syscall/handler/timerfd.rs

```diff
@@ -0,0 +1,244 @@
+            let remaining = match borrowed.get_timer_remaining() {
+                Some(simtime) => simtime,
+                None => SimulationTime::ZERO,
+            };
+            // We return a zero duration if the timer is non-periodic (set to expire only once).
+            let interval = borrowed.get_timer_interval();
+
+            (remaining, interval)
+        };
+
+        // Write the result to the plugin.
+        {
+            // Need a location to write the result.
+            if value_ptr.is_null() {
+                return Err(Errno::EFAULT);
+            }
+
+            // Error if the process disappeared.
+            let Some(runnable) = ctx.objs.process.borrow_runnable() else {
+                return Err(Errno::EINVAL);
+            };
+
+            // Set up the result values.
+            let result = libc::itimerspec {
+                it_value: remaining.try_into().unwrap(),
+                it_interval: interval.try_into().unwrap(),
+            };
+
+            // Write the result.
+            runnable.memory_borrow_mut().write(value_ptr, &result)?;
+        }
+
+        Ok(())
+    }
+
+    #[log_syscall(/* rv */ libc::c_int, /* fd */ libc::c_int, /* flags */ libc::c_int,
+                  /* new_value */ *const libc::c_void, /* old_value */ *const libc::c_void)]
+    pub fn timerfd_settime(
+        ctx: &mut SyscallContext,
+        fd: libc::c_int,
+        flags: libc::c_int,
+        new_value_ptr: ForeignPtr<libc::itimerspec>,
+        old_value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> SyscallResult {
+        let Some(flags) = TimerSetTimeFlags::from_bits(flags) else {
+            log::warn!("Invalid timerfd_settime flags: {}", flags);
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Get the TimerFd object.
+        let file = get_cloned_file(ctx, fd)?;
+        let File::TimerFd(ref timerfd) = file else {
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Read in the new value from the plugin.
+        let new_value = {
+            // Need to be able to read the new timer config.
+            if new_value_ptr.is_null() {
+                return Err(Errno::EFAULT.into());
+            }
+
+            // Error if the process disappeared.
+            let Some(runnable) = ctx.objs.process.borrow_runnable() else {
+                return Err(Errno::EINVAL.into());
+            };
+
+            // this is a clippy false-positive
+            #[allow(clippy::let_and_return)]
+            let val = runnable.memory_borrow().read(new_value_ptr)?;
+            val
+        };
+
+        // Verify a valid range for new_time nanosecond vals.
+        if new_value.it_value.tv_nsec < 0
+            || new_value.it_value.tv_nsec >= 999_999_999
+            || new_value.it_interval.tv_nsec < 0
+            || new_value.it_interval.tv_nsec >= 999_999_999
+        {
+            return Err(Errno::EINVAL.into());
+        }
```

Comment:
Optional: Instead of these checks, we could do the conversion to `SimulationTime` here instead. For example:

```rust
let value: SimulationTime = new_value.it_value.try_into().or(Err(Errno::EINVAL))?;
let interval: SimulationTime = new_value.it_interval.try_into().or(Err(Errno::EINVAL))?;

...

if value.is_zero() {
    ...
```

The `try_into` conversions already check the values of `tv_sec` and `tv_nsec`. (The existing checks here don't check of negative `tv_sec` values.)